build-system/tasks/bundle-size.js

```diff
@@ -32,7 +32,11 @@ const {
 const {getStdout} = require('../exec');
 const {gitCommitHash, gitTravisMasterBaseline} = require('../git');
 
-const runtimeFile = './dist/v0.js';
+const CORE_RUNTIME_FILE = './dist/v0.js';
+
+const INABOX_RUNTIME_FILE = './dist/amp4ads-v0.js';
+const INABOX_RUNTIME_GOLDEN_SIZE = 70; // In kilobytes
```

Comment:
Ack on the concern. I agree that forcing everyone to rebase would be bad, so I ran some numbers:

Now 70.06KB
8/15 71.03KB
8/1 70.36KB
7/15 75.65KB
7/1 70.46KB
6/15 75KB
6/1 74.8KB
5/15 74.74KB

The numbers are indeed quite stable. I can think of two things here:
* We could set up to update the number weekly, or whenever a major change commits
* We could change it not to block the PR, but only to notify us.

Obviously we also prefer the proper way. These are all stop gap solutions indeed.

Reply:
I really really really really really really hated the old system of a maximum build size. IT WAS TERRIBLE, and made me super frustrated every time a build blocked.

Reply:
@danielrozenberg How much work is it to expand the current v0.js check to cover more files? Can we massage sizes of extra binaries to the same DB field as serialized KV pairs? That way avoid a DB schema update.

Reply:
+1 to @danielrozenberg's instinct and @jridgewell's objection. Adding a field to the size reported to the bot is the way to go, even if it involves more work up front. We can do this for more than just the inbox binary.

Reply:
As @powerivq mentioned, can we make this temp solution non PR blocking . So at least we have some alert before a better solution is available?

Reply:
If all you want is a weekly alert, I think writing a cron job that fetches map4ads-v0.js and inspects its size once a week is better than introducing more complexity to this script :)

Right now the "DB schema" is actually this GitHub repository: https://github.com/ampproject/amphtml-build-artifacts/tree/master/bundle-size - in retrospect this wasn't the best choice and I should have made this a real DB schema so we could more easily perform updates (we could still transition to a real DB but this isn't a high priority item for me)

Reply:
@danielrozenberg Having this run per PR will make it easier to find the commit in question. Or maybe we can print all file sizes on travis on master runs? That would help too.

Reply:
> If all you want is a weekly alert, I think writing a cron job that fetches map4ads-v0.js and inspects its size once a week is better than introducing more complexity to this script :)

We want the alert as early as possible because such a regression might not be trivial to fix, and we don't want it be detected last minute by the release script.  Detecting the regression after release is not acceptable as it might already caused ads revenue drop.

> Right now the "DB schema" is actually this GitHub repository: 

I think storing those numbers in files are completely fine, I honestly don't see the need for a complicated DB at this moment. We just need to change the string format:
from `85.23KB` to `v0.js:85.23KB,amp4ads-v0.js:70.23KB`

All the parsing and alert triggering logic can be in `amp-github-app`. 

Would that work?

Reply:
In theory yes, but what you're describing is begging for a proper db at this point :)

There was also a request to start tracking the sizes of extensions (#20843). @rsimha do you think it might be worth prioritizing this type of work?

Reply:
In theory yes, but what you're describing is begging for a proper db at this point :)

There was also a request to start tracking the sizes of extensions (#20843). @rsimha do you think it might be worth prioritizing this type of work?

Reply:
> In theory yes, but what you're describing is begging for a proper db at this point :)

Not really. DB is only useful if you are going to query it in different ways. Otherwise it's just extra maintenance cost, such as DB schema migration, DB software update etc. 

That said, I actually like that you stored them in files keying by commit SHA, as I see we only need to read the whole blob of data for that commit at a time and operate on it.

Unless you have a bigger scope for this.

Reply:
I agree that a DB is likely too heavyweight for our purpose. Converting the file to json with keys for different files might be easiest.

Thinking about it some more: The json format can include the name of the file, the target bundle size, and maybe a boolean for whether to fail on increase or not (if you think it's necessary). The GitHub app will have to support the old and new format for a while, and modify the way it applies statuses. Then we can add size tracking for new files fairly easily. 

Obviously, this is just one of many options. @danielrozenberg WDYT?

Also, if there are multiple requests to track sizes of different things, it might be worth solving them in one fell swoop.

Reply:
+1 with one suggestion: we can leave alert config out of the storage.

Reply:
Circling back on this: @danielrozenberg @rsimha Do we want to move on with changing Github "DB"? That sounds to me a modest change and it should not impede any of the refactoring work you have in mind.

Reply:
Adding a new field to the file saved in the artifacts repo (after converting it to json) SGTM. Will let @danielrozenberg provide final approval since he knows this code best.

Reply:
I strongly disagree with this approach - this feels like technical debt in the making, trying to solve a single issue where there's demand for a more generic solution. I feel like the best approach here is to scope the work in a way that will solve both the requirement for amp4ads and for extensions and take it from there.

If there's a time constraint, we should figure out a quick discardable hack until the full solution is in place